### DISCUSSION TITLE
Java agent not fully support Tanuki Service Wrapper module?

### DISCUSSION THREAD

**Discussion by nvc97** [apache/skywalking#7876] [Q&A]
I'm implementing java agent into a Tanuki wrapper module but there're problems when I'm trying to view my service on UI:
- Does not show metrics for Service tab. (pic 1)
- Does not show Service Instance Load, Service Instance Throughput, Service Instance Successful Rate, Service Instance Latency in Instance tab. (pic 2)
- Cannot detect endpoint.
- Every Trace Segment only have 1 Span. (pic 3)

Previously, I've installed java agent in Tomcat modules to test it out, and it worked OK.

![1](https://user-images.githubusercontent.com/56123257/136544350-9687f703-863c-4a9e-9820-b8ef5d330a0d.jpeg)

![2](https://user-images.githubusercontent.com/56123257/136544360-047838bb-1b78-470b-b90e-ff50b7a09afa.jpeg)

![3](https://user-images.githubusercontent.com/56123257/136544356-67f8fead-6158-46e9-b085-3dae5dc71fcc.jpeg)

**Comment by wu-sheng** [comment]
I have no knowledge of Tanuki. 
What do you mean implementing agent? Is that a plugin or something?

**Reply by nvc97** [reply]
Well I mean that I'm installing java agent to Tanuki wrapper just similar to installing on Tomcat (`"CATALINA_OPTS=-javaagent:/path/to/skywalking-agent/skywalking-agent.jar"`)

For Tanuki wrapper, I just edit the `wrapper.conf` file with:
`wrapper.java.additional.6=-javaagent:/absolute/path/to/skywalking-agent/skywalking-agent.jar`

**Reply by wu-sheng** [reply]
I think we don't have this plugin. Please check document first.

**Reply by nvc97** [reply]
Oh, there's no plugin for Tanuki wrapper in the supported list. Would you consider adding it in future releases?

**Reply by wu-sheng** [reply]
SkyWalking plugins are contributed by community. Typically, we don't know what and when new plugin is added.
